mobile/src/main/java/org/openhab/habdroid/ui/PreferencesActivity.java

```diff
@@ -322,13 +346,39 @@ protected void updateAndInitPreferences() {
                 getParent(fullscreenPreference).removePreference(fullscreenPreference);
             }
 
-            if (!CloudMessagingHelper.isSupported()
-                    || Build.VERSION.SDK_INT >= Build.VERSION_CODES.O) {
+            if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.O) {
                 Log.d(TAG, "Removing notification prefs");
                 getParent(ringtonePref).removePreference(ringtonePref);
                 getParent(vibrationPref).removePreference(vibrationPref);
             }
 
+            if (Build.VERSION.SDK_INT < Build.VERSION_CODES.LOLLIPOP) {
+                Log.d(TAG, "Removing alarm clock prefs");
+                getPreferenceScreen().removePreference(alarmClockPrefCat);
+            } else {
+                updateAlarmClockEnabledPreferenceIcon(alarmClockEnabledPref,
+                        getPreferenceBool(alarmClockEnabledPref, false));
+                alarmClockEnabledPref.setOnPreferenceChangeListener((preference, newValue) -> {
+                    updateAlarmClockEnabledPreferenceIcon(preference, newValue);
+                    BackgroundTasksBroadcastReceiver.startAlarmChangedWorker(getActivity(),
+                            getPreferenceString(PREFERENCE_ALARM_CLOCK_ITEM, ""),
```

Comment:
The default value specified here doesn't match the `android:defaultValue` for that pref in XML.

Reply:
The pref is saved when the app is started: https://github.com/openhab/openhab-android/blob/master/mobile/src/main/java/org/openhab/habdroid/ui/MainActivity.java#L174. So it can never be non-existent and the defValue is never used, right?

Reply:
TIL I learned about that API ... I'll take a look into its implementation tomorrow.

Reply:
As `readAgain` is set to false there, I don't think it covers any prefs that were added after the user opened the app for the first time. If it's set to true I can confirm the default values are put into the shared prefs map, so yes, defValue is not used in that case. Setting it to true will probably slightly delay app startup, though...

Reply:
Maybe run that call with `readAgain` set to true in `OnUpdateBroadcastReceiver.updateComparableVersion()` instead if the version stored in prefs doesn't match `BuildConfig.VERSION_CODE`?

Reply:
I installed a version without this patch, started the app, applied this patch and started the app again. The default value will be set as soon as I open the preference activity. W/o opening the prefs the user cannot enable this feature, so using `""` as default seems save to me.

Reply:
> The default value will be set as soon as I open the preference activity

How exactly did you check/verify that?

Reply:
OK, fair enough. I still think I'd prefer doing it explicitly on update, but I can also make a separate PR for that.

Reply:
Yes, it should also load the local and remote settings.